libexec/scoop-import.ps1

```diff
@@ -0,0 +1,176 @@
+# Usage: scoop import <filepath/filepaths> [options]
+# Summary: Import exported list!
+# Help: The basic way to install exported list:
+#      scoop import exportedfile.txt
+# It also supports for multiple file:
+#      scoop import file1.txt file2.txt
+
+function createAppsList($filePaths) {
+    $appList = New-Object System.Collections.ArrayList
+    if ($filePaths) {
+        foreach ($file in $filePaths) {
+            if (checkFileExt($file)) {
+                foreach ($line in (Get-Content $file | Select-Object -Unique)) {
+                    if (checkEmptyLine($line)) {
+                        if (checkComment($line)) {
+                            continue
+                        }
+                        else {
+                            $appList += ($line + "`r`n")
+                        }
+                    }
+                }
+            }
+            else {
+                $extError = "Please select .txt file!"
+                throw $extError
+            }
+        }
+    }
+    else {
+        $filePathsError = "There is no file to exported!"
+        throw $filePathsError
+    }
+
+    $uniqueAppList = $appList | Get-Unique
+
+    return $uniqueAppList
+}
+
+function checkFileExt ($file) {
+    $ext = [IO.Path]::GetExtension($file)
+    if ($ext -eq ".txt") {
+        return $True
+    }
+    else {
+        return $false
+    }
+}
+
+function checkEmptyLine ($line) {
+    if ($line -ne "") {
+        return $True
+    }
+    else {
+        return $False
+    }
+}
+function checkComment ($line) {
+    if ($line.StartsWith("#") -Or $line.StartsWith(";") -Or $line.StartsWith(" ")) {
+        return $True
+    }
+    else {
+        return $False
+    }
+}
+function checkGlobal ($line) {
+    if ($line.contains("*global*")) {
+        return $True
+    }
+    else {
+        return $False
+    }
+}
+
+function checkArch ($line) {
+    if ($line.contains("{32bit}")) {
+        return $True
+    }
+    else {
+        return $False
+    }
+}
+
+function checkURL ($line) {
+    if ($line.contains(".json")) {
+        return $True
+    }
+    else {
+        return $False
+    }
+}
+
+function importApps($appslist) {
+
+
+    $appslist | ForEach-Object {
+
+        $globalArgs = ""
+        $archArgs = ""
+
+        if (checkAppsInstalled($_)) {
+            if (checkURL($_)) {
+                if (checkGlobal($_)) {
+                    $globallArgs += "--global "
+                }
+                if (checkArch($_)) {
+                    $archlArgs += "--arch 32bit "
+                }
+
+                $_ -match '\[(?<url>\S+.json)\]' > $null
+
+                try {
+                    scoop install $Matches.url $globalArgs.TrimEnd(" ") $archArgs.TrimEnd(" ")
+                }
+                catch {
```

Comment:
I cannot figure out how to catch if url not exist? If anybody have suggestions, I'm very happy to hear it!

Reply:
[super-julho-enigma-5918cf3ba9c24ff3e1e543798e55923952b1b2ab.zip](https://github.com/lukesampson/scoop/files/2432325/super-julho-enigma-5918cf3ba9c24ff3e1e543798e55923952b1b2ab.zip)

Reply:
You could check status code of the url `HEAD`

```
PS > Invoke-WebRequest -Uri 'community.spiceworks.com' -Method 'HEAD' | select StatusCode

StatusCode
----------
       200
```